src/main/java/com/imsweb/layout/LayoutUtils.java

```diff
@@ -48,6 +50,40 @@ private LayoutUtils() {
     }
 
     /**
+     * Reads an unspecified layout from the provided stream, expects XML format.
+     * Created on October 16, 2017 by schadega
+     * @param stream <code>InputStream</code> to the data file, cannot be null
+     * @return a Layout that could either be a <code>CommaSeparatedLayout</code> or <code>FixedColumnsLayout</code>, never null
+     * @throws IOException if the layout can't be read from the input stream
+     */
+    public static Layout readLayout(InputStream stream) throws IOException {
+        if (stream == null)
+            throw new IOException("Unable to read layout, target input stream is null");
+        try {
+            XStream xstream = new XStream(new StaxDriver() {
+                @Override
+                public HierarchicalStreamWriter createWriter(Writer out) {
```

Comment:
This code to create an instance of XStream is repeated four times now; can you please add a method that creates it so the code is not repeated (guess the method can be public, doesn't hurt). I think the call to the constructor and the setting the annotation auto-detection should be part of that method; the call to set the alias should stay with the individual methods.

Reply:
I saw you used your new "createXstream method in the three other read method, but you should also use it in this method.